packages/eslint-plugin/lib/rules/no-extra-spacing-text.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @typedef { import("../types").RuleModule } RuleModule
+ * @typedef { import("../types").ProgramNode } ProgramNode
+ * @typedef { import("es-html-parser").CommentContentNode } CommentContentNode
+ * @typedef { import("../types").ContentNode } ContentNode
+ * @typedef { import("../types").TextNode } TextNode
+ */
+
+const { RULE_CATEGORY } = require("../constants");
+
+const MESSAGE_IDS = {
+  UNEXPECTED: "unexpected",
+};
+
+/**
+ * @type {RuleModule}
+ */
+module.exports = {
+  meta: {
+    type: "code",
+
+    docs: {
+      description: "Disallow unnecessary consecutive spaces",
+      category: RULE_CATEGORY.BEST_PRACTICE,
+      recommended: false,
+    },
+
+    fixable: true,
+    schema: [
+      {
+        type: "object",
+        properties: {
+          skip: {
+            type: "array",
+            items: {
+              type: "string",
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      [MESSAGE_IDS.UNEXPECTED]: "Multiple consecutive spaces not allowed here",
+    },
+  },
+
+  create(context) {
+    const options = context.options[0] || {};
+    const skipTags = options.skip || [];
+    const sourceCode = context.getSourceCode();
+
+    /**
+     * @param {Array<ContentNode>} siblings
+     */
+    function checkSiblings(siblings) {
+      for (
+        let length = siblings.length, index = 0;
+        index < length;
+        index += 1
+      ) {
+        const node = siblings[index];
+
+        if (node.type === `Tag` && skipTags.includes(node.name) === false) {
+          checkSiblings(node.children);
+        } else if (node.type === `Text`) {
+          stripConsecutiveSpaces(node);
+        } else if (node.type === `Comment`) {
+          stripConsecutiveSpaces(node.value);
+        }
+      }
+    }
+
+    return {
+      Program(node) {
+        // @ts-ignore
+        checkSiblings(node.body);
+      },
+    };
+
+    /**
+     * @param {TextNode | CommentContentNode} node
+     */
+    function stripConsecutiveSpaces(node) {
+      const text = node.value;
+      const matcher = /(^|[^\n \t])([ \t]{2,})($|[^\n \t])/g;
```

Comment:
This regex doesn't seem to match trailing spaces. Is it possible to handle trailing whitespace as well?
```js
// spaces exists after the span
    {
      code: `
<div>
   <span></span>  
</div>
      `,
      output: `
<div>
   <span></span>
</div>
      `,
      errors: errorsAt([3, 16, 2]),
    },
  ],
```
<img width="232" alt="스크린샷 2024-09-04 오전 1 11 02" src="https://github.com/user-attachments/assets/f2387def-c44c-4434-90d3-d212fa8a77a3">

Reply:
@yeonjuan  Added!